### PR TITLE
feat: add page up and page down handling to row DOM performance demo

### LINKED ISSUE #46: Row DOM Performance - page up / down doesn't work

### PR DESCRIPTION
**Pull request by steve-haar**
fixes #46

### CODE REVIEW

File: grid-demo/src/app/angular/row-dom-performance/custom-clr-virtual-rows.directive.ts [hunk 1]
```diff
@@ -112,6 +112,8 @@ export class CustomClrVirtualRowsDirective<T> implements OnInit, DoCheck, OnDest
     this.updateFixedSizeVirtualScrollInputs();
   }
 
+  @Input('customClrKeyboardScrollPageSize') keyboardScrollPageSize = 10;
```
Comment (kevinbuhmann) [inline grid-demo/src/app/angular/row-dom-performance/custom-clr-virtual-rows.directive.ts:115]: Structural directive inputs should be prefixed with the directive name. So this one should be `customClrVirtualRowsKeyboardScrollPageSize`.

Comment (kevinbuhmann) [inline grid-demo/src/app/angular/row-dom-performance/custom-clr-virtual-rows.directive.ts:115]: Structural directive inputs should be prefixed with the directive name. So this one should be `customClrVirtualRowsKeyboardScrollPageSize`.

File: grid-demo/src/app/angular/row-dom-performance/row-dom-performance.component.html [hunk 2]
```diff
@@ -30,9 +30,11 @@
     <ng-container *clrDgHideableColumn>{{ column.displayName }}</ng-container>
   </clr-dg-column>
 
-  <clr-dg-row *customClrVirtualRows="let vm of data.vms" [clrDgItem]="vm">
-    <clr-dg-cell *ngFor="let column of columns">{{ vm[column.field] }}</clr-dg-cell>
-  </clr-dg-row>
+  <ng-template customClrVirtualRows let-vm [customClrVirtualRowsOf]="data.vms" [customClrKeyboardScrollPageSize]="100">
+    <clr-dg-row [clrDgItem]="vm">
```
Comment (kevinbuhmann) [inline grid-demo/src/app/angular/row-dom-performance/row-dom-performance.component.html:34]: You don't have to de-sugar this. `<clr-dg-row *customClrVirtualRows="let vm of data.vms; keyboardScrollPageSize: 100" [clrDgItem]="vm">...</clr-dg-row>`

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]

File: grid-demo/src/app/angular/row-dom-performance/custom-clr-virtual-rows.directive.ts [hunk 3]
```diff
@@ -210,6 +221,16 @@ export class CustomClrVirtualRowsDirective<T> implements OnInit, DoCheck, OnDest
   private restoreActiveCellAfterRenderedRangeUpdate() {
     if (this.activeCellElement && this.activeCellElement.tabIndex === 0) {
       setTimeout(() => {
+        this.updateAriaRowIndexes();
```
Comment (kevinbuhmann) [inline grid-demo/src/app/angular/row-dom-performance/custom-clr-virtual-rows.directive.ts:224]: Is this a separate bug fix?

Comment (steve-haar) [reply grid-demo/src/app/angular/row-dom-performance/custom-clr-virtual-rows.directive.ts:224]: no, it's only necessary for this feature

File: grid-demo/src/app/angular/row-dom-performance/custom-clr-virtual-rows.directive.ts (hunk 1 above)
Comment (steve-haar) [reply grid-demo/src/app/angular/row-dom-performance/custom-clr-virtual-rows.directive.ts]: fixed

File: grid-demo/src/app/angular/row-dom-performance/row-dom-performance.component.html (hunk 2 above)
Comment (steve-haar) [reply grid-demo/src/app/angular/row-dom-performance/row-dom-performance.component.html]: I didn't know it worked that way. Fixed

Comment (kevinbuhmann) [reply grid-demo/src/app/angular/row-dom-performance/row-dom-performance.component.html]: All good!

**Review by kevinbuhmann — APPROVED** [review]

**Pull request merged**